### ISSUE TITLE
Make session timeout configurable

### ISSUE THREAD

**Issue by dacrome** [osiam/osiam#76]
Moved from https://github.com/osiam/auth-server/issues/7:

@tkrille wrote:

> 30 minutes is not enough for most of the use cases. Session timeout MUST be configurable.

@dacrome wrote:

> This fruit hangs so low, you just need to grap it from the ground. With Spring Boot it's now configurable with [`server.session-timeout`](http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html). The property just need to be documented and added to the [example configuration file](https://github.com/osiam/osiam/blob/4745cafee0ad6f9637b413333d07b353f717f019/src/main/deploy/osiam.properties).

**Comment by wallner** [comment]
It's not as low hanging as assumed at first so we reschedule it for 3.1

**Comment by dacrome** [comment]
I do not remember, what is the problem with the session configuration? I guess we need to persist the session data between restarts.

**Comment by tkrille** [comment]
The problem is that, while we can easily configure this for standalone deployment, when running OSIAM in Tomcat, one must configure the Tomcat to change session timeouts. But there's also [Spring Session](http://projects.spring.io/spring-session/), which seems to completely take care of the session management, when activated.

I think we should start with projecting the Spring Boot property `server.session.timeout` onto the `osiam` namespace, e.g. `osiam.session.timeout` and adding a very short explanation for Tomcat deployments to the documentation with links to the Tomcat docs. Before that, somebody should verify the stuff I wrote in the first paragraph :)

**Comment by dacrome** [comment]
I did not test it, but I think you are right, the `server.session.timeout` is only working with the standalone configuration. Let's do it like you described.

**Comment by dacrome** [comment]
I think this is obsolete since we want to persist the sessions with Spring Session (#293) and then we can or have to set the timeouts globally @wallner @tkrille

**Comment by tkrille** [comment]
But we still have to make it configurable. Particularly, creating a new configuration property, e.g. `osiam.session.timeout`.

**Comment by dacrome** [comment]
okay, but I think the config is an implicit task of the persist sessions feature.

**Comment by tkrille** [comment]
I don't think so. After #305 is merged, we will just obtain the ability to store sessions in the database. That's the scope of the associated task, #293. We will again use the default value for session timeout, which is 30 Minutes.
